tests/Common/ComponentsEventsTest.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace Keboola\Test\Common;
+
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Options\Components\Configuration;
+use Keboola\StorageApi\Options\Components\ConfigurationRow;
+use Keboola\StorageApi\Options\Components\ListComponentsOptions;
+use Keboola\Test\StorageApiTestCase;
+
+class ComponentsEventsTest extends StorageApiTestCase
+{
+    const COMPONENT_ID = 'wr-db';
+    const CONFIGURATION_NAME = 'test';
+
+    /**
+     * @var int
+     * generated uid
+     */
+    private $configurationId;
+
+    /**
+     * @var Components
+     */
+    private $components;
+
+    /**
+     * @var string
+     */
+    private $tokenId;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->components = new Components($this->_client);
+        foreach ($this->components->listComponents() as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $this->components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        // erase all deleted configurations
+        foreach ($this->components->listComponents((new ListComponentsOptions())->setIsDeleted(true)) as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $this->components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+        $this->configurationId = $this->_client->generateId();
+
+        $this->components->addConfiguration($this->getConfiguration());
+        $this->tokenId = $this->_client->verifyToken()['id'];
+    }
+
+    /**
+     * @return Configuration
+     */
+    private function getConfiguration()
+    {
+        $config = (new Configuration())
+            ->setComponentId(self::COMPONENT_ID)
+            ->setConfigurationId($this->configurationId)
+            ->setName(self::CONFIGURATION_NAME);
+        return $config;
+    }
+
+    public function testConfigurationChange()
+    {
+        $config = $this->getConfiguration();
+        $config->setDescription('new desc');
+        $this->components->updateConfiguration($config);
+        $events = $this->listEvents();
+        self::assertEquals('storage.componentConfigurationChanged', $events[0]['event']);
+        self::assertEquals('Changed component configuration test (wr-db)', $events[0]['message']);
+    }
+
+    public function testConfigurationCreate()
+    {
+        // check create event
+        $events = $this->listEvents();
+        self::assertEquals('storage.componentConfigurationCreated', $events[0]['event']);
+        self::assertEquals('Created component configuration test (wr-db)', $events[0]['message']);
```

Comment:
Všude názvy konfigurací do uvozovek, aby to bylo poznat, kdyby to bylo prázdné nebo nějaký netisknutelný znak (`Created component configuration "test" (wr-db)`)